src/modules/vendor/components/courses/ProfileFollowUp.vue

```diff
@@ -63,6 +102,60 @@ export default {
 
       return missingInfo;
     },
+    isFinished () {
+      const slots = this.course.slots
+        .filter(slot => this.$moment().isBefore(slot.startDate))
+      return !slots.length;
+    },
+  },
+  methods: {
+    openSmsModal () {
+      this.updateMessage();
+      this.opened = true;
+    },
+    updateMessage () {
+      const courseLink = `${location.protocol}//${location.hostname}${(location.port ? ':' + location.port : '')}/trainees/courses/${this.course._id}`;
+      if (this.typeMessage === 'Convocation') {
+        const slots = this.course.slots.sort((a, b) => a.startDate - b.startDate);
+        const date = this.$moment(slots[0].startDate).format('DD/MM/YYYY');
+        const hour = this.$moment(slots[0].startDate).format('HH:mm');
+
+        this.messageComp = `Bonjour,\nVous êtes inscrits à la formation ${this.course.name}.\nLa première session à ` +
+        `lieu le ${date} à partir de ${hour}.\nMerci de vous présenter au moins 15 minutes avant le début de la ` +
+        `formation.\nToutes les informations sur : ${courseLink}\nNous vous souhaitons une bonne formation,\nCompani`;
+      } else if (this.typeMessage === 'Reminder') {
+        const slots = this.course.slots
+          .filter(slot => this.$moment().isBefore(slot.startDate))
+          .sort((a, b) => a.startDate - b.startDate);
+        const date = this.$moment(slots[0].startDate).format('DD/MM/YYYY');
+        const hour = this.$moment(slots[0].startDate).format('HH:mm');
+
+        this.messageComp = `Bonjour,\nRAPPEL : vous êtes inscrits à la formation ${this.course.name}.\nVotre ` +
+        `prochaine session à lieu le ${date} à partir de ${hour}.\nMerci de vous présenter au moins 15 minutes avant ` +
+        `le début de la formation.\nToutes les informations sur : ${courseLink}\nNous vous souhaitons une bonne ` +
+        'formation,\nCompani'
+      };
+    },
+    async sendMessage () {
```

Comment:
en fait la je ferai plus une route `course/courseID/send-sms` et c'est dans le back qu'on gere la boucle for
et comme ca au niveau des droits c'est `courses:edit`
qu'en penses tu ? 
et @jctrebalag ?

Reply:
je suis d'accord et j'ajouterais que tu pourrais utiliser un `Promise.all` pour envoyer tous ces messages en même tps :)